hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/events/HmppsDomainEventsEmitterIntTest.kt

```diff
@@ -0,0 +1,572 @@
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.boot.test.mock.mockito.SpyBean
+import org.springframework.data.repository.findByIdOrNull
+import software.amazon.awssdk.services.sns.SnsAsyncClient
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest
+import software.amazon.awssdk.services.sqs.model.Message
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.DiffCategory.IDENTIFIERS
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.DiffCategory.LOCATION
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.SyncIndex
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.PrisonerBuilder
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.readResourceAsText
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.repository.PrisonerHashRepository
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.services.PrisonerDifferenceService
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.wiremock.PrisonApiExtension.Companion.prisonApi
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.wiremock.PrisonApiMockServer
+import uk.gov.justice.hmpps.sqs.MissingQueueException
+import uk.gov.justice.hmpps.sqs.PurgeQueueRequest
+import uk.gov.justice.hmpps.sqs.countAllMessagesOnQueue
+
+class HmppsDomainEventsEmitterIntTest : IntegrationTestBase() {
+  @SpyBean
+  private lateinit var hmppsDomainEventEmitter: HmppsDomainEventEmitter
+
+  @Autowired
+  private lateinit var objectMapper: ObjectMapper
+
+  @SpyBean
+  @Qualifier("offenderqueue-sqs-client")
+  private lateinit var offenderQueueSqsClient: SqsAsyncClient
+
+  @SpyBean
+  @Qualifier("hmppseventtopic-sns-client")
+  private lateinit var hmppsEventTopicSnsClient: SnsAsyncClient
+
+  @SpyBean
+  private lateinit var prisonerDifferenceService: PrisonerDifferenceService
+
+  @Autowired
+  private lateinit var prisonerHashRepository: PrisonerHashRepository
+
+  private val hmppsEventsQueue by lazy {
+    hmppsQueueService.findByQueueId("hmppseventtestqueue")
+      ?: throw MissingQueueException("hmppseventtestqueue queue not found")
+  }
+
+  @BeforeEach
+  fun purgeHmppsEventsQueue() = runTest {
+    with(hmppsEventsQueue) {
+      hmppsQueueService.purgeQueue(PurgeQueueRequest(queueName, sqsClient, queueUrl))
+    }
+  }
+
+  @BeforeEach
+  fun init() {
+    prisonApi.stubOffenders(PrisonerBuilder())
+    buildAndSwitchIndex(SyncIndex.GREEN, 1)
+  }
```

Comment:
I imagine this slows the tests down somewhat? I wonder if we can do this a single time and just refresh the single record before each test?